packages/client/src/actions/applyAction.ts

```diff
@@ -202,3 +209,28 @@ async function remapBatchActionParams<
 
   return remappedParams;
 }
+
+function remapActionResponse(
+  response: SyncApplyActionResponseV2 | BatchApplyActionResponseV2,
+): ActionEditResponse | undefined {
+  const editResponses = response?.edits;
+  if (editResponses?.type === "edits") {
+    const { edits, ...editResponse } = editResponses;
+    const remappedActionResponse: ActionEditResponse = {
+      ...editResponse,
+      addedLinks: [],
+      addedObjects: [],
+      modifiedObjects: [],
+    };
+    for (const edit of editResponses.edits) {
+      if (edit.type === "addLink") {
+        remappedActionResponse.addedLinks.push(edit);
+      } else if (edit.type === "addObject") {
+        remappedActionResponse.addedObjects.push(edit);
+      } else if (edit.type === "modifyObject") {
+        remappedActionResponse.modifiedObjects.push(edit);
+      }
```

Comment:
Do we need an assertNever here?

Reply:
It wouldn't hurt. Or an `else { invariant(false, "Unknown edit type") }`.